### PR TITLE
Add static constructors, remove global instance, change linter

### PR DESCRIPTION
**Pull request by h3llofr1end**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -45,9 +45,9 @@ mem:
 	go tool pprof --alloc_space pprof/fastime-test.bin pprof/mem-fastime.out
 
 lint:
-	gometalinter --enable-all . | rg -v comment
+	golangci-lint run
```
Comment (sashayakovtseva) [inline Makefile:48]: if you switch to golangci, maybe add config as well?

File: fastime.go [hunk 2]
```diff
@@ -47,6 +47,29 @@ func New() Fastime {
...
+	return f
+}
+
+// NewStaticWithFormat returns new Fastime that is initialized with the given time and format.
+// Returned instance is not updated automatically.Call Fastime.StartTimerD
+// to start update in the background. This function is useful in testing.
+func NewStaticWithFormat(t time.Time, format string) Fastime {
+	f := newFastime()
+	f.SetLocation(t.Location()).
+		SetFormat(format)
```
Comment (sashayakovtseva) [inline fastime.go:67]: do not chain please

**Review by sashayakovtseva — CHANGES_REQUESTED** [review]

**Comment by sashayakovtseva** [comment]
Can you rename this PR to reflect changes more explicit? E.g. `Add static constructors, remove global instance`

File: Makefile (hunk 1 above)
Comment (h3llofr1end) [reply Makefile:48]: added

File: fastime.go (hunk 2 above)
Comment (h3llofr1end) [reply fastime.go]: remove chain call

File: example/main.go [hunk 3]
```diff
@@ -2,9 +2,8 @@ package main
 
 import (
 	"fmt"
+	"github.com/viru-tech/fastime/v2"
 	"time"
```
[the remark this answers is not in the document]
Comment (h3llofr1end) [reply example/main.go]: add prefix github.com to golangci.yml

File: .golangci.yml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    enable-all: true
+  golint:
+    min-confidence: 0.8
+  gofmt:
+    simplify: true
+  gci:
+    sections:
+      - standard
+      - default
+      - prefix(github.com)
```
Comment (sashayakovtseva) [inline .golangci.yml:19]: github.com/viru-tech

File: .golangci.yml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+      linters: [ errorlint ]
+    - path: "_test\\.go"
+      linters:
+        - bodyclose
+        - dupl
+        - funlen
+        - goconst
+        - gosec
+        - noctx
+        - wrapcheck
```
Comment (sashayakovtseva) [inline .golangci.yml:105]: missing LF

File: .gitignore [hunk 6]
```diff
@@ -10,3 +10,5 @@
 
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
+
+.idea
```
Comment (proxeter) [inline .gitignore:14]: Add also .vscode please

File: Makefile [hunk 7]
```diff
@@ -45,9 +45,9 @@ mem:
...
 
-test: clean
+test:
 	GO111MODULE=on go test --race -v $(go list ./... | rg -v vendor)
```
Comment (proxeter) [inline Makefile:51]: You can remove GO111MODULE=on
It's by default already

File: .golangci.yml (hunk 4 above)
Comment (h3llofr1end) [reply .golangci.yml]: fixed

File: .gitignore (hunk 6 above)
Comment (h3llofr1end) [reply .gitignore:14]: added

File: Makefile (hunk 7 above)
Comment (h3llofr1end) [reply Makefile]: removed

**Review by sashayakovtseva — APPROVED** [review]

**Pull request merged**